src/js/libgeo/GeoOps.js

```diff
@@ -1724,12 +1737,68 @@ geoOps.TrMoebiusP.kind = "P";
 geoOps.TrMoebiusP.updatePosition = function(el) {
     var t = csgeo.csnames[(el.args[0])];
     var p = csgeo.csnames[(el.args[1])].homog;
-    var l1 = List.productMV(t.mat1, p);
-    var l2 = List.productMV(t.mat2, p);
-    el.homog = List.normalizeMax(List.cross(l1, l2));
+
+    el.homog = geoOps._helper.TrMoebiusP(p, t.mat1, t.mat2);
     el.homog = General.withUsage(el.homog, "Point");
 };
 
+geoOps._helper.TrMoebiusP = function(p, mat1, mat2) {
+    var l1 = List.productMV(mat1, p);
+    var l2 = List.productMV(mat2, p);
+    return List.normalizeMax(List.cross(l1, l2));
+};
+
+geoOps._helper.TrMoebiusPts = function(tr, a1, a2, a3) {
+    var b1 = geoOps._helper.TrMoebiusP(a1, tr.mat1, tr.mat2);
+    var b2 = geoOps._helper.TrMoebiusP(a2, tr.mat1, tr.mat2);
+    var b3 = geoOps._helper.TrMoebiusP(a3, tr.mat1, tr.mat2);
+    return [b1, b2, b3];
+};
+
+geoOps.TrMoebiusL = {};
+geoOps.TrMoebiusL.kind = "C";
+geoOps.TrMoebiusL.updatePosition = function(el) {
+    var t = csgeo.csnames[(el.args[0])];
+    var l = csgeo.csnames[(el.args[1])].homog;
+
+    var getRandLine = function() {
+        var rline = List.realVector([100 * (Math.random() - 0.5), 100 * (Math.random() - 0.5), 100 * (Math.random() - 0.5)]);
```

Comment:
Why the factor 100? I don't see that affecting the geometric meaning in any way, and I can't imagine it improving numerics either.